src/Field/Object_.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+// vim:ts=4:sw=4:et:fdm=marker:fdl=0
+
+namespace atk4\data\Field;
+
+use atk4\data\Field;
+use atk4\data\ValidationException;
+
+/**
+ * Object field type. Think of it as field type "object" in past.
+ */
+class Object_ extends Field
+{
+    /** @var string Field type for backward compatibility. */
+    public $type = 'object';
+
+    /**
+     * Validate and normalize value.
+     *
+     * @param mixed $value
+     *
+     * @throws ValidationException
+     *
+     * @return mixed
+     */
+    public function normalize($value)
+    {
+        if ($value === null || $value === '') {
+            if ($this->required) {
```

Comment:
parent call? not_null..

Reply:
not_null will be another extra property we already have mandatory/system/required that are still hard to understand at first, i suggest to hardcode in a parent call or in a specific protected method like : 

```php
protected function notNull($value) : void {
    if ((null === $value || $value === '') && $this->required) {
        throw new ValidationException([$this->name => 'Must not be null or empty']);
    }
}
```